scripts/gen-api-gtkdoc.py

```diff
@@ -396,13 +399,25 @@ def main(args):
     else:
         outfile = sys.stdout
 
+    if (opts.scioutfile):
+        try:
+            scioutfile = open(opts.scioutfile, "w+")
+        except OSError as err:
+            sys.stderr.write("failed to open \"%s\" for writing (%s)\n" % (opts.scioutfile, err.strerror))
+            return 1
+    else:
+        scioutfile = outfile
+
     try:
         outfile.write("/*\n * Automatically generated file - do not edit\n */\n\n")
-        outfile.write("#include <glib.h>\n")
-        outfile.write("#include <gtk/gtk.h>\n\n")
-        outfile.write("typedef struct _ScintillaObject ScintillaObject;\n")
+        outfile.write("#include \"gtkcompat.h\"\n")
+        outfile.write("#include \"sciwrappers.h\"\n")
```

Comment:
Am 9. März 2016 19:35:54 MEZ, schrieb Colomban Wendling <notifications@github.com>:
>>      try:
>>          outfile.write("/*\n * Automatically generated file - do not
>edit\n */\n\n")
>> -        outfile.write("#include <glib.h>\n")
>> -        outfile.write("#include <gtk/gtk.h>\n\n")
>> -        outfile.write("typedef struct _ScintillaObject
>ScintillaObject;\n")
>> +        outfile.write("#include \"gtkcompat.h\"\n")
>> +        outfile.write("#include \"sciwrappers.h\"\n")
>
>Why include sciwrappers here?  wouldn't it make more sense to include
>`Scintilla.h` and `ScintillaWidget.h`?
>Anyway, good point including *gtkcompat.h* for the GtkComboBoxText
>hack.  Though, now we depend on 2.24 I'd like to break API and ABI and
>drop the compat here at some point.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/geany/geany/pull/948/files#r55565475

Convenience, mostly. g-ir-scanner doesnt scam includes but uses type information from them if they are used in the main file. Only symbols defined in the files listed in the cmdline are picked up.